src/python/pants/task/target_restriction_mixins.py

```diff
@@ -23,6 +23,19 @@ class HasTransitiveOptionMixin:
 
   @property
   def act_transitively(self):
+    deprecated_conditional(
+      lambda: self.get_options().is_default("transitive"),
+      removal_version="1.25.0.dev2",
+      entity_description="Pants defaulting to `--fmt-transitive` and `--lint-transitive`",
+      hint_message="Pants will soon default to `--no-fmt-transitive` and `--no-lint-transitive`. "
+                   "Currently, Pants defaults to `--fmt-transitive` and `--lint-transitive`, which "
+                   "means that tools like isort and Scalafmt will work on transitive dependencies "
+                   "as well. This behavior is unexpected. Normally when running tools like isort, "
+                   "you'd expect them to only work on the files you specify.\n\nTo prepare, "
+                   "please add to your `pants.ini` under both the `fmt` and the `lint` "
+                   "sections the option `transitive: False`. If you want to keep the default, use "
+                   "`True`, although the option will be removed in Pants 1.27.0.dev2."
```

Comment:
Is there an inconsistency between this 1.27... and the 1.25... at line 28 or am I missing something?

Reply:
There are two deprecations. This one deprecates the default value, and then the next deprecation will remove the feature entirely. I see why that could be confusing, though. It's probably better to just say "in the future, we will remove this option".